bundles/org.openhab.core.model.thing/src/org/openhab/core/model/thing/internal/GenericThingProvider.xtend

```diff
@@ -614,9 +614,9 @@ class GenericThingProvider extends AbstractProviderLazyNullness<Thing> implement
                 createThing(newThings, factory)
             ]
         }
-        thingsMap.put(modelName, newThings)
 
         newThings.forEach [ newThing |
+            thingsMap.get(modelName).add(newThing)
```

Comment:
Is `thingsMap.get(modelName)` guaranteed to be non-null? I see the method is also called from `thingHandlerFactoryAdded` .

Reply:
I think yes, because when called from `thingHandlerFactoryAdded` the model names are retrieved from the keys of the `thingsMap` and the `thingsMap` has non-null values.